### ISSUE TITLE
DB changes for notifications (SID-33)

### ISSUE THREAD

**Issue by bjmckenz** [csufmse/student_information_system#256]

**Comment by bjmckenz** [comment]
Database support and message list in /schooladmin present.
did not include message detail (view), or message send pages.

Messages are sent from one user to another. They have a date sent, a date read, and a date archived (deleted). They have a priority (high or normal). The have a body and subject, and an "in response to" field.

from a Profile, access them via "sent_to" and "sent_by". Note that the views I implemented in /schooladmin do a hard filter on "time_archived__isnull=True" so that the Prof and Stud doesn't see "archived" messages. The Admins do though.

There are tables for each of sent and received. They *bold* unread messages and /italicize/ archived messages.

Test data is present. Most students sent a message or two to an admin each semester. Admins almost always read them and responded. Students did not always read those.

Students occasionally got high-priority alerts saying they were in trouble,

**NOTE: two new modules were added to requirements.txt. Be sure to install them.**

Testing:
* from /schooladmin/users, view admins -- ares, for example.
* go to /schooladmin/student with the message sender and see that the message is there too.
* Explore filtering and sorting.

Known issues (UI):
* filter sets include more fields than appropriate: archived and sent by (for sent messages) and received by (for received messages). Those should be removed.
* obviously we need a click on message -> detail, which marks it Read (setting time_read to now()).
* New message too.
